addons/project_purchase/models/project_project.py

```diff
@@ -15,7 +15,7 @@ class Project(models.Model):
     @api.depends('analytic_account_id')
     def _compute_purchase_orders_count(self):
         data = self.env['purchase.order.line']._read_group(
-            [('analytic_distribution', 'in', self.analytic_account_id.ids)],
+            [('analytic_distribution', 'in', self.analytic_account_id.ids), ('company_id', 'in', self.env.companies.ids)],
```

Comment:
I am not sure to understand why we need to do that since we have a ir.rule for that. 🤯 

https://github.com/odoo/odoo/blob/65094a3af7aaea4f03e9eed89fb03a93df50dd02/addons/purchase/security/purchase_security.xml#L49-L53